core/src/main/java/hudson/node_monitors/DfRunner.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016, Baptiste Mathus
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package hudson.node_monitors;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.logging.Logger;
+
+class DfRunner {
+	private static final Logger LOGGER = Logger.getLogger(DfRunner.class.getSimpleName());
+
+	private static Map<String, DfCommand> IMPLEMENTATIONS = new LinkedHashMap<String, DfCommand>();
+
+	static {
+		IMPLEMENTATIONS.put("windows", new WindowsDfCommand());
+		IMPLEMENTATIONS.put("linux", new LinuxDfCommand());
+		IMPLEMENTATIONS.put("mac os", new MacOsDfCommand());
+		IMPLEMENTATIONS.put("freebsd", new MacOsDfCommand()); // Same as Mac
+		IMPLEMENTATIONS.put("aix", new AixDfCommand());
+	}
+
+	public String getUsedInodesPercentage() {
```

Comment:
The implementation is right in your code.
My proposal was to document the behavior in order to avoid misusages of the new public API by other core/plugin developers